modules/dxp/apps/saml/saml-opensaml-integration/src/main/java/com/liferay/saml/opensaml/integration/internal/credential/FileWatcher.java

```diff
@@ -120,33 +120,38 @@ public FileWatcher(
 				}
 
 				List<WatchEvent<?>> watchEvents = watchKey.pollEvents();
+				List<WatchEvent<Path>> watchEventsPath = new ArrayList<>();
 
-				Stream<Path> pathsStream = _paths.stream();
-				Stream<WatchEvent<?>> watchEventsStream = watchEvents.stream();
+				for (Path path : _paths) {
+					for (WatchEvent<?> watchEvent : watchEvents) {
+						WatchEvent<Path> watchEventPath =
+							(WatchEvent<Path>)watchEvent;
 
-				Stream<WatchEvent<Path>> watchEventsPathStream =
-					pathsStream.flatMap(
-						path -> watchEventsStream.map(
-							watchEvent -> (WatchEvent<Path>)watchEvent
-						).filter(
-							watchEvent -> {
-								Path contextPath = watchEvent.context();
+						Path contextPath = watchEventPath.context();
 
-								return contextPath.endsWith(path.getFileName());
-							}
-						));
+						if (!contextPath.endsWith(path.getFileName())) {
+							continue;
+						}
+
+						watchEventsPath.add(watchEventPath);
+					}
+				}
+
+				CompletableFuture[] completableFutures =
+					new CompletableFuture[watchEventsPath.size()];
+
+				for (int i = 0; i < completableFutures.length; i++) {
+					int finalI = i;
```

Comment:
@jeffwu0724 Instead of using a "final" version of variable `i`, you can add an SF commit to get the `Path` from the previous list using `i`, and use the `Path` variable in the lambda:

```
WatchEvent<Path> watchEventPah = watchEventsPath.get(i);

Runnable runnable = () -> _consumer.accpet(watchEventPath);
```